docs/Developers/NextGen_Core.rst

```diff
@@ -0,0 +1,303 @@
+NextGen-Core
+============
+
+This document provides a brief overview of coala's NextGen-Core.
+coala's NextGen-Core comes with the promise of lifting many limitations of the
+existing core and better efficiency and performance.
+
+What is new?
+------------
+
+The following new features have been added as a part of the NextGen-Core:
+
+- Easier Interface
+- Official support for virtual files
+- Improved dependency system
+- New base bear type: ``DependencyBear``
+- Ability to modify bear dependencies at runtime
+- Custom parallelization for bears
+- Superior caching
+
+What has changed?
+-----------------
+
+- ``Global Bears`` are now called ``Project Bears`` and ``Local Bears`` are
+  now known as ``File Bears``. Both the ``ProjectBear`` and the ``FileBear``
+  classes inherit from the base class ``coalib.core.Bear``.
```

Comment:
The `Bear` class' definition has also been changed. Maybe you can mention that too and then change this line to 
```
... inherit from the base class ``Bear``
```